indexing-service/src/test/java/org/apache/druid/indexing/overlord/TaskQueueTest.java

```diff
@@ -690,6 +690,52 @@ public TaskStatus runTask(TaskToolbox toolbox)
     Assert.assertEquals(failedStatus, getTaskStorage().getStatus(task.getId()).get());
   }
 
+  @Test
+  public void testTaskWaitingTimeMetricNotEmittedWhenTaskNotReady() throws Exception
+  {
+    // task1 acquires a lock that will block task2
+    final TestTask task1 = new TestTask("t1", Intervals.of("2021-01/P1M"));
+    prepareTaskForLocking(task1);
+    Assert.assertTrue(task1.isReady(actionClientFactory.create(task1)));
+
+    // task2 will not be ready because of task1's lock
+    final TestTask task2 = new TestTask("t2", Intervals.of("2021-01-31/P1M"));
+    taskQueue.add(task2);
+    taskQueue.manageQueuedTasks();
+
+    Thread.sleep(100);
```

Comment:
(not required in this PR) I think we should start using the `LatchableEmitter` for these testing needs so that we can remove the sleeps from the unit tests.